Library/Services/ContextStorage.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+namespace Vpg\Disturb\Services;
+
+use Vpg\Disturb\Exceptions\ContextStorageException;
+use Vpg\Disturb\ContextStorageAdapters\ContextStorageAdapterInterface;
+
+use \Phalcon\Config\Adapter\Json;
+
+class ContextStorage
+{
+    /**
+     * @var ContextStorageAdapterInterface $adapter
+     */
+    private $adpater;
+
+    /**
+     * ContextStorage constructor
+     *
+     * @param Json $config
+     *
+     * @throws ContextStorageException
+     */
+    public function __construct(Json $config)
+    {
+        // check adapter type
+        if (empty($config->adapter)) {
+            throw new ContextStorageException(
+                'Adapter name not found',
+                ContextStorageException::CODE_ADAPTER
+            );
+        }
+
+        // check if adapter class exists
+        $adapterClass = ucfirst($config->adapter) . 'Adapter';
+        if (! class_exists($adapterClass)) {
+            throw new ContextStorageException(
+                'Adapter class not found',
+                ContextStorageException::CODE_ADAPTER
+            );
+        }
+
+        // check if adapter config exists
+        if (empty($config->config)) {
+            throw new ContextStorageException(
+                'Adapter config not found',
+                ContextStorageException::CODE_ADAPTER
+            );
+        }
+
+        $this->adpater = new $adapterClass();
```

Comment:
should be abstract, the client config give us an adapter CODE from which we decide which class instantiate